### PR TITLE
Updating deployment role for 1.21

### PR DESCRIPTION
**Pull request by austinbyers**
## Background
Copying internal changes we made to the deployment role to reduce its scope.

## Changes
- Reduced scope of permissions for DynamoDB, ECR, S3, SNS, SQS, Lambda, KMS, and IAM

## Testing
1.21.0-RC root-stack deployment with this role

### CODE REVIEW

**Review by s0l0ist — COMMENTED** [review]
Let's hold off on merging until 1.21 is GA.

**Review by nhakmiller — APPROVED** [review]

**Comment by nhakmiller** [comment]
Approved so it'll be ready to merge when the time comes, but we can still wait for 1.21 to be in GA before merging.

Seems like CI is blocked as well for all PRs in this repo, not sure what that's about but might be a permissions issue.

**Comment by austinbyers** [comment]
I've updated the role with permissions that work for all deployments 1.18+, so it's safe to merge now

**Comment by austinbyers** [comment]
I think I've fixed CircleCI for future runs, but for now I just had to admin override

**Pull request merged**
